fdbcli/LocationMetadataCommand.actor.cpp

```diff
@@ -53,27 +78,25 @@ ACTOR Future<Void> printRandomShards(Database cx, int n, bool physicalShard) {
 			try {
 				tr->setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
 				tr->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
+				tr->setOption(FDBTransactionOptions::LOCK_AWARE);
 
 				state RangeResult UIDtoTagMap = wait(tr->getRange(serverTagKeys, CLIENT_KNOBS->TOO_MANY));
 				ASSERT(!UIDtoTagMap.more && UIDtoTagMap.size() < CLIENT_KNOBS->TOO_MANY);
 
-				KeyRangeRef currentKeys(begin, allKeys.end);
-				RangeResult shards = wait(krmGetRanges(tr, keyServersPrefix, currentKeys, n, CLIENT_KNOBS->TOO_MANY));
+				state KeyRangeRef currentKeys(begin, allKeys.end);
+				state RangeResult shards =
+				    wait(krmGetRanges(tr, keyServersPrefix, currentKeys, n, CLIENT_KNOBS->TOO_MANY));
 
-				for (int i = 0; i < shards.size() - 1 && numShards < n; ++i) {
-					const KeyRangeRef currentRange(shards[i].key, shards[i + 1].key);
+				state int i = 0;
+				for (; i < shards.size() - 1 && numShards < n; ++i) {
+					KeyRangeRef currentRange(shards[i].key, shards[i + 1].key);
 					std::vector<UID> src;
 					std::vector<UID> dest;
 					UID srcId;
 					UID destId;
 					decodeKeyServersValue(UIDtoTagMap, shards[i].value, src, dest, srcId, destId);
```

Comment:
Do we still need this decode? Seems that printKeyServersEntry does the decode as well.